eventlet/hubs/asyncio.py

```diff
@@ -24,6 +32,12 @@ class Hub(hub.BaseHub):
 
     def __init__(self):
         super().__init__()
+
+        # Make sure asyncio thread pools use real threads:
+        if concurrent_imported:
+            concurrent.futures.thread.threading = original("threading")
```

Comment:
1. `spawn_for_awaitable` having no monkey-patching would be a bunch of work, involving thread local access on every API call. I am not sure it's actually necessary, either: anything that runs in `async` context is supposed to be non-blocking, and eventlet only patches blocking APIs.
2. The `select` patching issue is covered by #934. Once that is merged I will return to this PR.

Reply:
> As the hubs are event loops made to dispatch I/O events and to schedule threads, so in this case I think it make sense to unpatch threading/queue at the hub level.

This is not quite correct. Threads are _not_ scheduled by the hub. Rather, pretty much every Python event loop tends to have an internal thread pool at absolute minimum, for doing DNS lookups, because `socket.gethostbyname()` and friends are blocking APIs. You can't run blocking APIs in the event loop thread, so you need a thread pool. Thus Twisted has `reactor.callInThread()`, and here we have `asyncio.to_pool()`.

Given that internal asyncio thread pool exists, and is _specifically designed_ for blocking APIs, we want _at minimum_ that internal thread pool to continue to use normal threads and queues.

This particular PR goes further and allows _all_ users of `concurrent.futures.thread` thread pools to use normal thread pools, on the assumption that existing eventlet code doesn't use that particular API with green threads. That assumption may be wrong... so if you find this concerning I can try to make it specific to asyncio's internal thread pool, but that might be much more difficult and brittle.

Reply:
>     2. The `select` patching issue is covered by [Run tests on macOS in CI, and some fixes to get it in reasonable state #934](https://github.com/eventlet/eventlet/pull/934). Once that is merged I will return to this PR.

Done.

Thanks for previous details and observations.